blog-posts/guardian-bundle-size/guardian-bundle-size.md

```diff
@@ -0,0 +1,64 @@
+---
+published: false
+title: 'Best practice (1): control the size of your Lambda function bundle'
+cover_image:
+description: ''
+tags: serverless, lambda, quality, AWS
+series:
+canonical_url:
+---
+
+_This article is part of a series on [Guardian][guardian], an open-source, highly configurable, automated best-practice audit tool for Serverless architectures._
+
+# The lighter, the better
+
+A light bundle will positively impact your Serverless architecture for at least two reasons.
+
+## 1. A light deployment package reduces the cold start of your Lambda function
+
+The Serverless Framework automatically bundles your code and uploads it to AWS. The underlying tools are [serverless-webpack][serverless-webpack] or [serverless-esbuild][serverless-esbuild], which should be preferred for its shorter configuration and better performance.
+
+Not only does the size impact the time it takes to upload the bundle, it also significantly impacts the cold start.
+
+![Cold start durations](./assets/bundle-size-impact-on-cold-start.png 'Cold start durations per deployment size (https://mikhail.io/serverless/coldstarts/aws/#does-package-size-matter)')
+
+## 2. AWS enforces quotas on deployment resources
+
+Smaller bundles will help you respect [AWS quotas][quotas].
+
+| Resource                                          | Quota  |
+|---------------------------------------------------|--------|
+| Lambda function deployment package (zipped)       | 50 MB  |
+| Lambda function deployment package (uncompressed) | 250 MB |
+| Total deployment packages                         | 75 GB  |
+
+
+# Guardian is your solution to easily check all your bundles
+
+[Guardian][guardian] now offers a **new rule** to warn you when uploading bundles over 5 MB. This is a demanding rule because this threshold can easily be reached. Gather your team and decide what deployment bundles should be limited to.
+
+Guardian comes with **many rules** to help you make the best decisions for your Serverless project. Guardian will help you identify where your code can be optimized to achieve better performance at a lower cost.
```

Comment:
```suggestion
Guardian comes with **many rules** to help you make the best decisions for your Serverless project. It will help you identify where your code can be optimized to achieve better performance at a lower cost.
```